Step 1

NB993 opened the pull request:
[step-1 ]단어 밀어내기

NB993 committed:
[Add] InputView 생성

'단어' '방향' '밀어낼 길이'를 한 줄로 입력받을 클래스.

NB993 committed:
[Add] InputView 유효성 검사 추가

1. 공백 없는 단어가 입력되었는지
2. 밀어낼 방향을 올바르게 입력했는지
3. 밀어낼 횟수를 올바르게 입력했는지
4. 각 한 칸씩만 띄웠는지

NB993 committed:
[Fix] validateCount수정

코드 위치만 변경

NB993 committed:
[Fix] InputView validate메서드 수정

- validateLength : 길이가 3이 아닌 경우 예외처리. 매직넘버 사용.
- validateDirection : toUpperCase -> equalsIgnoreCase로 변경

NB993 committed:
[Add] Word 단어 밀어내기 기능 추가

방향과 밀어낼 카운트를 입력받아 리스트를 자른 후 조합하여 출력

NB993 committed:
[Add] InputView scanner close

Main에서 단어 출력 후 inputview의 scanner를 종료

NB993 committed:
[Fix] Main 클래스 input()제거

main메서드로 이동

NB993 committed:
[Fix] make RIGHT_INPUT_LENGTH static variable

NB993 committed:
[Fix] Word 메서드 분리

단어 밀어내기 메서드 분리

jeremy0405 commented on src/step_1/InputView.java:
```diff
@@ -1,5 +1,62 @@
+    }
+  }
+
+  private void validateWord(String word) throws IllegalArgumentException {
+    if (word.contains(" ")) {
+      throw new IllegalArgumentException("공백 없는 단어를 입력해주세요.");
+    }
+  }
```
input을 trim() 이후 spilt(" ") 으로 이미 나눠놨는데 이 부분이 과연 의미가 있을까요??

NB993 replied:
아...!!!!

jeremy0405 commented on src/step_1/Main.java:
```diff
@@ -2,4 +2,14 @@
+  public static void main(String[] args) {
+    InputView inputView = new InputView();
+    String[] input = inputView.input();
+    String inputWord = input[0];
+    String direction = input[1];
+    int count = Integer.parseInt(input[2]);
+
+    Word word = new Word(inputWord, direction, count);
```
굳이 메인에 inputWord, direction, count가 있어야 할까요?
`String[] input` 을 Word에 생성자로 바로 넣는게 좋지 않을까요??

NB993 replied:
뭘 넣는지 명시적으로 보여주려고 했는데 생각해보니 Word 클래스 생성자에서 확인할 수가 있었네요. 제리 말씀대로 불필요한 코드같습니당.

jeremy0405 commented on src/step_1/Word.java:
```diff
@@ -1,5 +1,65 @@
+
+  private void pushRightPlus(int pushCount) {
+    List<String> pushedAlphabets = new ArrayList<>();
+    pushedAlphabets.addAll(alphabets.subList(alphabets.size() - pushCount, alphabets.size()));
+    pushedAlphabets.addAll(alphabets.subList(0, alphabets.size() - pushCount));
```
`alphabets.size()` 를 많이 사용하는데
`int size = alphabets.seze()` 선언을 해줘서 `size`로 대체하는게 좋을 것 같습니다 :)

NB993 replied:
무지성으로 size()를 남발하고 있었네요.. 감사합니다!

NB993 committed:
[Remove] InputView validateWord()

validateLength()에서 이미 공백에 대한 예외처리를 했는데
validateWord()에서 공백 체크를 중복수행하고 있었음.

NB993 committed:
[Update] Main.java parameter for constructor of Word

Word클래스 생성자와 pushAlphabet메서드에서 String[] input의
각 요소들이 어떤 값들인지를 확인할 수 있으므로 Main에서는
배열을 통째로 넘겨주기만 하는 방향으로 변경.

NB993 committed:
[Update] use local variable of wordLength

alphabets.size()를 생각없이 남발하고 있어서 지역변수인
wordLength를 사용하도록 변경.
